### ISSUE TITLE
Update plugins howto to cover new API for <package>/plugins/__init__.py added in #2339

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2769]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#2769|
|Type|defect|
|Created|2007-08-02 15:31:29Z|

<details><summary>Searchable metadata</summary>

```
trac-id__2769 2769
type__defect defect
reporter__itamarst itamarst
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1186068689000000 1186068689000000
changetime__1187453170000000 1187453170000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

I'm not sure I understand. Are you talking about twisted.plugin.pluginPackagePaths? If that's the case, it's not a new API for me: it's just a new internal functions used to cleanly find plugins. That's not intended to be used by plugin developers.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

Sorry forget about last comment.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

I adjusted stuff in plugin-package-paths-doc-2769, please review.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_4"></a>|@dreid set owner to @therve|
|-|-|

What is a "simple directory"? Why not just say:

```
This directory must <em>not</em> be a Python package: including an __init__.py will cause the directory to be skipped and no plugins loaded from it.
```

Of course this change is only barely related to the topic of this ticket, and the reflowing of the formatted text makes this diff longer than necessary.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml commented|
|-|-|

Accidentally reviewed this ticket. I totally agree with dreid's suggestion.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @exarkun|
|-|-|

I am the author of this code.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @dreid|
|-|-|

I changed that sentence.

I'm not sure if you meant that the contents of the change are unrelated to this branch or if only the reflowing was.

The bug being fixed in the branch was precisely that python packages were having plugins loaded from them.  On the chance that someone was using this behavior, despite the fact that the documentation tried to explain that you should not, I decided it would be a good idea to make the documentation more explicit about this matter.

As for the reflowing, I agree.  However, I don't see what alternative exists now.  The diffs generated after edits to a single line containing an entire paragraph of text are similarly unreadable.

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_8"></a>|@dreid set owner to @exarkun|
|-|-|

I was refering to the reflowing, but if it couldn't be avoided, go ahead and merge it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [21090]) Merge plugin-package-paths-doc-2769

Author: exarkun
Reviewer: dreid, jml
Fixes [#2769](https://github.com/twisted/twisted/issues/2769)

Update example definition of a project's plugin package's __init__.py to use
the recently added twisted.plugin.pluginPackagePaths utility function.  Also make
it somewhat more explicit that only a single Python package will be examined for
plugins.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
